CoreFoundation/Base.subproj/CFBase.h

```diff
@@ -127,16 +127,20 @@
 #endif
 
 #if TARGET_OS_WIN32
-    #if !defined(CF_EXPORT)
-        #if defined(CF_BUILDING_CF) && defined(__cplusplus)
-            #define CF_EXPORT extern "C" __declspec(dllexport)
-        #elif defined(CF_BUILDING_CF) && !defined(__cplusplus)
-            #define CF_EXPORT extern __declspec(dllexport)
-        #elif defined(__cplusplus)
-            #define CF_EXPORT extern "C" __declspec(dllimport)
+    #if defined(__cplusplus)
+        #define _CF_EXTERN extern "C"
+    #else
+        #define _CF_EXTERN extern
+    #endif
+
+    #if defined(_USRDLL)
```

Comment:
This is defined by the Windows SDK or compiler when building shared libraries?

Reply:
Its a slight abuse of the normal behaviour for Windows.  `_USRDLL` is pretty commonly defined (by Visual Studio's default build rules), but it technically is meant for ATL/MFC applications building a COM server for a DLL (InProc server).